### PR TITLE
Update cloud-init to 21.3

### PR DESCRIPTION
**Pull request by jiria**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Updates cloud-init to 21.2. Refreshes patches on top of 21.2.

###### Change Log  

- Deleted CVE patches, as they have been merged upstream
- Rebased remaining patches

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- local build, vhd deployed and basic cloud init functionality validated, version update confirmed

### CODE REVIEW

File: SPECS/cloud-init/cloud-init.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
 %define python3_sitelib /usr/lib/python3.7/site-packages
 
 Name:           cloud-init
-Version:        19.1
-Release:        6%{?dist}
+Version:        21.2
+Release:        0%{?dist}
```
Comment (anphel31) [inline SPECS/cloud-init/cloud-init.spec:5]: we've been starting all releases at 1; could you do the same?

**Comment by anphel31** [comment]
sounds like the check tests are still passing; can you confirm?

**Review by anphel31 — APPROVED** [review]

File: SPECS/cloud-init/cloud-init.spec (hunk 1 above)
Comment (jiria) [reply SPECS/cloud-init/cloud-init.spec]: Fixed.

**Comment by jiria** [comment]
> sounds like the check tests are still passing; can you confirm?

If you are referring to this line:

Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files

Then imho yes.

To validate, I ran: `sudo make build-packages CONFIG_FILE= RUN_CHECK=y` from the MarinerDermo repo that has been initialized based on the public instructions from github. Was that sufficient?

**Comment by jiria** [comment]
Can you please have a look at the linter failure and tell me what is wrong there?

**Comment by anphel31** [comment]
> > sounds like the check tests are still passing; can you confirm?
> 
> If you are referring to this line:
> 
> Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
> 
> Then imho yes.
> 
> To validate, I ran: `sudo make build-packages CONFIG_FILE= RUN_CHECK=y` from the MarinerDermo repo that has been initialized based on the public instructions from github. Was that sufficient?

Yes that should be sufficient to run the tests. You can open the cloud-init build log (in build/logs/pkggen/rpmbuilding/cloud-init-*.log and you should see the cloud-init test output followed by:
"====== CHECK DONE cloud-init. EXIT STATUS 0"

**Comment by jiria** [comment]
> > > sounds like the check tests are still passing; can you confirm?
> > 
> > 
> > If you are referring to this line:
> > Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
> > Then imho yes.
> > To validate, I ran: `sudo make build-packages CONFIG_FILE= RUN_CHECK=y` from the MarinerDermo repo that has been initialized based on the public instructions from github. Was that sufficient?
> 
> Yes that should be sufficient to run the tests. You can open the cloud-init build log (in build/logs/pkggen/rpmbuilding/cloud-init-*.log and you should see the cloud-init test output followed by:
> "====== CHECK DONE cloud-init. EXIT STATUS 0"

Just reran the build to make sure, but I cannot find `CHECK` in that log. Any thoughts? Is it because I am building this in a derivative repo?

**Comment by anphel31** [comment]
> > > > sounds like the check tests are still passing; can you confirm?
> > > 
> > > 
> > > If you are referring to this line:
> > > Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
> > > Then imho yes.
> > > To validate, I ran: `sudo make build-packages CONFIG_FILE= RUN_CHECK=y` from the MarinerDermo repo that has been initialized based on the public instructions from github. Was that sufficient?
> > 
> > 
> > Yes that should be sufficient to run the tests. You can open the cloud-init build log (in build/logs/pkggen/rpmbuilding/cloud-init-*.log and you should see the cloud-init test output followed by:
> > "====== CHECK DONE cloud-init. EXIT STATUS 0"
> 
> Just reran the build to make sure, but I cannot find `CHECK` in that log. Any thoughts? Is it because I am building this in a derivative repo?

In that case the tests must not be running. You might need to force a rebuild with `sudo make build-packages CONFIG_FILE= RUN_CHECK=y PACKAGE_BUILD_LIST="cloud-init" PACKAGE_REBUILD_LIST="cloud-init"`. I tried the updated tests but am seeing some failures.

**Comment by jiria** [comment]
> In that case the tests must not be running. You might need to force a rebuild with `sudo make build-packages CONFIG_FILE= RUN_CHECK=y PACKAGE_BUILD_LIST="cloud-init" PACKAGE_REBUILD_LIST="cloud-init"`. I tried the updated tests but am seeing some failures.

Thanks Andrew, I can see the test run results now and the failures too. Will investigate further. Side question, should we add these notes into the Demo repo readme?

**Review by PawelWMS — APPROVED** [review]

**Comment by anphel31** [comment]
> > In that case the tests must not be running. You might need to force a rebuild with `sudo make build-packages CONFIG_FILE= RUN_CHECK=y PACKAGE_BUILD_LIST="cloud-init" PACKAGE_REBUILD_LIST="cloud-init"`. I tried the updated tests but am seeing some failures.
> 
> Thanks Andrew, I can see the test run results now and the failures too. Will investigate further. Side question, should we add these notes into the Demo repo readme?

Good idea, I'll update the Demo repo readme with check test information.

**Comment by jiria** [comment]
Closing in favor of https://github.com/microsoft/CBL-Mariner/pull/1405

**Pull request closed without merging**